src/core/infer_request.h

```diff
@@ -194,6 +194,27 @@ class InferenceRequest {
     std::map<std::string, std::shared_ptr<Memory>> host_policy_data_map_;
   };
 
+  // State tensor
```

Comment:
State itself shouldn't live inside a request, should it? A State is associated with a sequence and so likely should live inside the sequence batcher. It seems like a request should just have a reference to a "sequence object" and whenever need to get a state object it should go though that sequence object. For now the sequence object may just hold maps of input_name->State and output_name ->State

Reply:
The state temporarily lives inside the request before `TRITONBACKEND_UpdateState` is called. There are state slots in the `SequenceBatch` object that the state will be transferred to after `TRITONBACKEND_UpdateState` is called. I will modify this according to your comment.